### PR TITLE
[mongo-c-driver] update to 1.25.2 

### LINKED ISSUE #35506: mongo-c-driver update to 1.25.2
### Library name

mongo-c-driver

### New version number

1.25.2

### Other information that may be useful (release notes, etc...)

Release notes: https://github.com/mongodb/mongo-c-driver/releases/tag/1.25.2

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/35506
Realted to https://github.com/microsoft/vcpkg/pull/35521
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/mongo-c-driver/fix-dependencies.patch [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
- 
+@@ -334,10 +334,7 @@ if (USE_SYSTEM_LIBBSON)
+    # The input variable BSON_ROOT_DIR is respected for backwards compatibility,
+    # but you should use the standard CMAKE_PREFIX_PATH instead.
+    message (STATUS "Searching for libbson CMake packages")
+-   find_package (bson-1.0
+-      "${PROJECT_VERSION}"
+-      HINTS
+-      ${BSON_ROOT_DIR})
++   find_package(bson-1.0 CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/mongo-c-driver/fix-dependencies.patch:13]: Why do we have to patch this spot at all?

Comment (jimwang118) [reply ports/mongo-c-driver/fix-dependencies.patch:13]: Using the upstream code will prompt that `libson-1.0` cannot be found and a compilation error occurs.

Comment (dg0yt) [reply ports/mongo-c-driver/fix-dependencies.patch:13]: Given that both packages are from the same upstream, isn't it an indication of a port problem if upstream's `find_package` doesn't work?

File: versions/l-/libbson.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "versions": [
     {
-      "git-tree": "cefffe3ffeb41cff3720044ea92a8734c42dc31c",
```
Comment (JonLiu1993) [inline versions/l-/libbson.json:4]: Please keep the old "git-tree" and add new one

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**